// @nodegui/nodegui/dist/lib/QtEnums/WindowFrameSection/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WindowFrameSection = void 0;
var WindowFrameSection;
(function (WindowFrameSection) {
    WindowFrameSection[WindowFrameSection["NoSection"] = 0] = "NoSection";
    WindowFrameSection[WindowFrameSection["LeftSection"] = 1] = "LeftSection";
    WindowFrameSection[WindowFrameSection["TopLeftSection"] = 2] = "TopLeftSection";
    WindowFrameSection[WindowFrameSection["TopSection"] = 3] = "TopSection";
    WindowFrameSection[WindowFrameSection["TopRightSection"] = 4] = "TopRightSection";
    WindowFrameSection[WindowFrameSection["RightSection"] = 5] = "RightSection";
    WindowFrameSection[WindowFrameSection["BottomRightSection"] = 6] = "BottomRightSection";
    WindowFrameSection[WindowFrameSection["BottomSection"] = 7] = "BottomSection";
    WindowFrameSection[WindowFrameSection["BottomLeftSection"] = 8] = "BottomLeftSection";
    WindowFrameSection[WindowFrameSection["TitleBarArea"] = 9] = "TitleBarArea";
})(WindowFrameSection = exports.WindowFrameSection || (exports.WindowFrameSection = {}));
