// @nodegui/nodegui/dist/lib/QtEnums/GlobalColor/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GlobalColor = void 0;
var GlobalColor;
(function (GlobalColor) {
    GlobalColor[GlobalColor["white"] = 3] = "white";
    GlobalColor[GlobalColor["black"] = 2] = "black";
    GlobalColor[GlobalColor["red"] = 7] = "red";
    GlobalColor[GlobalColor["darkRed"] = 13] = "darkRed";
    GlobalColor[GlobalColor["green"] = 8] = "green";
    GlobalColor[GlobalColor["darkGreen"] = 14] = "darkGreen";
    GlobalColor[GlobalColor["blue"] = 9] = "blue";
    GlobalColor[GlobalColor["darkBlue"] = 15] = "darkBlue";
    GlobalColor[GlobalColor["cyan"] = 10] = "cyan";
    GlobalColor[GlobalColor["darkCyan"] = 16] = "darkCyan";
    GlobalColor[GlobalColor["magenta"] = 11] = "magenta";
    GlobalColor[GlobalColor["darkMagenta"] = 17] = "darkMagenta";
    GlobalColor[GlobalColor["yellow"] = 12] = "yellow";
    GlobalColor[GlobalColor["darkYellow"] = 18] = "darkYellow";
    GlobalColor[GlobalColor["gray"] = 5] = "gray";
    GlobalColor[GlobalColor["darkGray"] = 4] = "darkGray";
    GlobalColor[GlobalColor["lightGray"] = 6] = "lightGray";
    GlobalColor[GlobalColor["transparent"] = 19] = "transparent";
    GlobalColor[GlobalColor["color0"] = 0] = "color0";
    GlobalColor[GlobalColor["color1"] = 1] = "color1";
})(GlobalColor = exports.GlobalColor || (exports.GlobalColor = {}));
